### PR TITLE
Add extra case for URL escaping

### PR DESCRIPTION
**Pull request by masongup-mdsol**
This change lets the MAuth client signature verification handle fully escaped URL strings in addition to the already present cases. This is needed to support applications that put arbitrary object names in the path for GET requests.

### CODE REVIEW

**Comment by masongup-mdsol** [comment]
The unit tests that are failing with this revision were also failing in the develop branch. Let me know if there is a fix to merge, or if we are using a different workflow.

**Comment by ejennings-mdsol** [comment]
Can you write a spec that failed before the fix and passes now?

File: lib/mauth/client.rb [hunk 1]
```diff
@@ -351,6 +352,10 @@ def signature_valid!(object)
         object.attributes_for_signing[:request_url].gsub!('%2F', '/') # ...and then 'simply' decode the %2F's back into /'s, just like Euresource kind of does!
         expected_euresource_style_reencoding = object.string_to_sign(time: object.x_mws_time, app_uuid: object.signature_app_uuid)
 
+        #And do yet another reencoding
```
Comment (ejennings-mdsol) [inline lib/mauth/client.rb]: Please state why this is necessary

File: lib/mauth/client.rb [hunk 2]
```diff
@@ -361,7 +366,7 @@ def signature_valid!(object)
           raise InauthenticError, "Public key decryption of signature failed!\n#{$!.class}: #{$!.message}"
         end
         # TODO: time-invariant comparison instead of #== ?
```
Comment (ejennings-mdsol) [inline lib/mauth/client.rb]: The comment should probably be updated if `==` is not being used any more

**Comment by jcarres-mdsol** [comment]
@masongup-mdsol This is an opensource repo so be careful but can you give us an example on a API that requires this call?
Maybe the original API should not be serializing object this way so I'm curious to see it.

If the change is valid, then this PR will need a spec , version  update and changelog entry.

**Comment by jfeltesse-mdsol** [comment]
As a follow up to PR #3 I synced master with develop and removed the develop branch entirely. I changed the base branch of this PR to be master so there shouldn't be consequences here but please make sure that locally you checkout master, remove the develop branch and update your config accordingly if needs be.

**Comment by masongup-mdsol** [comment]
@jcarres-mdsol The issue is that some piece of software between one of our apps as deployed on AWS and the internet is partially un-escaping the route. For example, when the request path is `/v1/topics/test%25%26`, the  RequestAuthenticator middleware sees `/v1/topics/Test&%25`. To match the route that the request was signed with, we have to fully un-escape the route, and then re-escape it.

I will write up the spec and changelog entry and add to this PR tomorrow.

**Comment by jfeltesse-mdsol** [comment]
Sounds like a hack trying to workaround an issue with your apps, not something that mauth-client should take care or even know of, IMHO you should start with having a clear understanding of why `the internet is partially un-escaping the route` in the first place.

**Comment by masongup-mdsol** [comment]
After discussion among our team, we will abandon this PR and implement a workaround in our project, and will continue to investigate the underlying issue behind this.

**Pull request closed without merging**